packages/opentelemetry-exporter/src/index.ts

```diff
@@ -0,0 +1,211 @@
+ */
+
+/** Span Kind
+Default value. Indicates that the span is used internally.
+INTERNAL = 0,
+
+Indicates that the span covers server-side handling of an RPC or other
+remote request.
+SERVER = 1,
+
+Indicates that the span covers the client-side wrapper around an RPC or
+other remote request.
+CLIENT = 2,
+
+Indicates that the span describes producer sending a message to a
+broker. Unlike client and server, there is no direct critical path latency
+relationship between producer and consumer spans.
+PRODUCER = 3,
+
+Indicates that the span describes consumer receiving a message from a
+broker. Unlike client and server, there is no direct critical path latency
+relationship between producer and consumer spans.
+CONSUMER = 4
+*/
+
+const SPAN_KIND_INTERNAL = 0;
+const SPAN_KIND_SERVER = 1;
+const SPAN_KIND_CLIENT = 2;
+const SPAN_KIND_PRODUCER = 3;
+const SPAN_KIND_CONSUMER = 4;
+
+const SERVER_KIND = 'server';
+const CLIENT_KIND = 'client';
+const PRODUCER_KIND = 'producer';
+const CONSUMER_KIND = 'consumer';
+const INTERNAL_KIND = 'internal';
+
+/**
+ * Status Code
+ *
+ * Default 
+ * UNSET = 0,
+ * The operation has been validated by an Application developer or
+ * Operator to have completed successfully.
+ * OK = 1,
+ * The operation contains an error.
+ * ERROR = 2
+ */
+const STATUS_UNSET = 0;
+const STATUS_OK = 1;
+const STATUS_ERROR = 2;
+
+const statusAsString = {
+  2: 'ERROR',
+  1: 'OK',
+  0: 'UNSET'
+};
+
+export class InstanaExporter implements SpanExporter {
+
+  constructor(agentKey?: string, endpointUrl?: string) {
+
+    // If endpoint URL and agent key are not provided, we don't set them in an attempt to use env vars, if they were set
+    if (endpointUrl) {
+      process.env[instanaEndpointUrlEnvVar] = endpointUrl;
+    }
+
+    if (agentKey) {
+      process.env[instanaAgentKeyEnvVar] = agentKey;
+    }
+
+    instanaEnvironment.validate();
+    instanaBackendConnector.init();
+  }
+
+  export(spans: ReadableSpan[], resultCallback: (result: ExportResult) => void): void {
+    const validSpans = spans.map(this._transform.bind(this));
+    this._sendSpans(validSpans, resultCallback);
+  }
+
+  shutdown(): Promise<void> {
+    return Promise.resolve();
+  }
+
+  private _sendSpans(spans: any[], resultCallback: (result: ExportResult) => void) {
+    instanaBackendConnector.sendSpans(spans,  (err: any, _responseBody?: string) => {
+      let result: ExportResult = { code: ExportResultCode.SUCCESS };
+      if (err) {
+        console.log('error', err);
+        result = { code: ExportResultCode.FAILED, error: err };
+      }
+      resultCallback(result);
+    });
+  }
+
+  private _transform(span: ReadableSpan): any {
+
+    let result: { [key: string]: any } = {
+      n: 'otel',
+      f: {
+        e: process.pid
+      },
+      data: {}
+    };
+
+    const traceId = this._buildTraceOrSpanId(span.spanContext().traceId, true);
+
+    if (traceId.length === 32) {
+      result.t = traceId.substr(16);
```

Comment:
We need to set the additional attribute `span.lt` when we limit the trace ID from 32 characters to 16 characters; we need to preserve the original long trace ID in `span.lt` (*l*ong *t*race ID), see here https://github.com/instana/agent/blob/master/agent-optional-parent/agent-open-telemetry/src/main/java/com/instana/agent/main/impl/otel/OpenTelemetryTraceEndpoint.java#L137-L138 and https://github.com/instana/nodejs/blob/main/packages/core/src/tracing/tracingHeaders.js#L364-L370.

Reply:
True. I missed that info when I started "converting" the Java code into JS.
Thanks!